### PR TITLE
add task queue management functionality

### PR DESCRIPTION
**Pull request by CaptainOfHacks**
* implement Task
* implement TaskQueue
* implement TaskManager
* implement TaskManager RestAPI

### CODE REVIEW

**Review by kaleanych — APPROVED** [review]

File: mapping_workbench/backend/task_manager/adapters/task.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+from datetime import datetime
+from enum import Enum
+from typing import Callable, Optional, List
+
+from pebble import ProcessFuture
+from pydantic import BaseModel
+
+
+class TaskStatus(str, Enum):
```
Comment (costezki) [inline mapping_workbench/backend/task_manager/adapters/task.py:9]: This looks like reinventing the wheel. Surely Celery does all this better and many more other things.

File: mapping_workbench/backend/task_manager/adapters/task.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    """
+    QUEUED = "QUEUED"
+    RUNNING = "RUNNING"
+    FINISHED = "FINISHED"
+    TIMEOUT = "TIMEOUT"
+    FAILED = "FAILED"
+    CANCELED = "CANCELED"
+
+
+class TaskResult:
```
Comment (costezki) [inline mapping_workbench/backend/task_manager/adapters/task.py:21]: Looks fairly clean, a line of documentation would not be redundant, however.

File: mapping_workbench/backend/task_manager/adapters/task.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    task_name: str
+    task_timeout: Optional[float]
+    task_status: TaskStatus
+    created_at: datetime
+    started_at: datetime = None
+    finished_at: datetime = None
+    exception_message: str = None
+
+
+class APIListTaskMetadataResponse(BaseModel):
```
Comment (costezki) [inline mapping_workbench/backend/task_manager/adapters/task.py:45]: Why is "API" involved at this level?

File: mapping_workbench/backend/task_manager/adapters/task.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        except Exception as e:
+            task_result.exception_message = str(e)
+            task_result.task_status = TaskStatus.FAILED
+        task_result.finished_at = datetime.now()
+        return task_result
+
+
+class Task:
+    """
+
```
Comment (costezki) [inline mapping_workbench/backend/task_manager/adapters/task.py]: starting this documentation line was a good intention

File: mapping_workbench/backend/task_manager/entrypoints/__init__.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+from mapping_workbench.backend.task_manager.adapters.task_manager import TaskManager
+
+AppTaskManager = TaskManager(max_workers=10)
```
Comment (costezki) [inline mapping_workbench/backend/task_manager/entrypoints/__init__.py:3]: This limitation looks arbitrary, we could configure it somewhere.

**Review by costezki — APPROVED** [review]
Looks like a clean implementation.
Not sure the existent tests suffice.

File: mapping_workbench/backend/task_manager/adapters/task.py (hunk 4 above)
Comment (CaptainOfHacks) [reply mapping_workbench/backend/task_manager/adapters/task.py]: You look at old version ) review again)

File: mapping_workbench/backend/task_manager/entrypoints/__init__.py (hunk 5 above)
Comment (CaptainOfHacks) [reply mapping_workbench/backend/task_manager/entrypoints/__init__.py]: I change this config to be provided from project settings.

File: mapping_workbench/backend/task_manager/adapters/task.py (hunk 2 above)
Comment (CaptainOfHacks) [reply mapping_workbench/backend/task_manager/adapters/task.py:21]: Documentation is already added.

File: mapping_workbench/backend/task_manager/adapters/task.py (hunk 1 above)
Comment (CaptainOfHacks) [reply mapping_workbench/backend/task_manager/adapters/task.py:9]: Yes, but this implementation is more what we need:
- in memory queue
- single node
- without additional services
- simple task management
- without inter-node communication
- simple workers without additional build settings
- easy configurable workers: ProcessPool or ThreadPool

**Pull request merged**